R/FilteredDataset.R

```diff
@@ -320,6 +343,21 @@ FilteredDataset <- R6::R6Class( # nolint
             logger::log_trace("FilteredDataset$srv_active@1 removed all non-anchored filters, dataname: { dataname }")
           })
 
+          observeEvent(input$show_new_filter_ui, {
+            shinyjs::show("filter_add_ui")
+            shinyjs::toggle("show_new_filter_ui")
+            shinyjs::toggle("hide_new_filter_ui")
+          })
+
+          observeEvent(input$hide_new_filter_ui, {
+            shinyjs::hide("filter_add_ui")
+            shinyjs::toggle("show_new_filter_ui")
+            shinyjs::toggle("hide_new_filter_ui")
+          })
+
+          self$srv_add(private$dataname)
```

Comment:
We need to keep the feature of not showing add panel when in `teal_slices` is specified with `allow_add = FALSE`.

```suggestion
          if (<teal_slices allow_add> == TRUE>) {
          self$ui_add(ns("add"), active_datanames)
        }
```

Reply:
I see that the `allow_add` completely hid the "Add filters" panel before. However, now it is possible to selectively hide the `+` button allowing the user to disable addition for selective datasets.
So, what if `allow_add` took dataset names as input. When not provided all datasets can be added, but, when provided, only those datasets can be added?

Reply:
> So, what if allow_add took dataset names as input

@vedhav let's not introduce breaking changes. `allow_add` has specific use cases and it came from end users.